test/integration/model/TestCenterServiceTest.php

```diff
@@ -59,13 +59,20 @@ public function testAssignUser()
         $this->assertEquals($assignedTc->getUri(), $this->tc->getUri());
     }
 
+    public function testUnassignUserException()
+    {
+        $this->expectException(TestCenterException::class);
+        $service = $this->getService();
+        $user = $this->getUserMock('proctor', $service);
```

Comment:
It should be tested with `admin` role as well as in the Service the `getAssigneeProperty()` handles this role as well.